### ISSUE TITLE
Integrating Jazz Cloud with rclone

### ISSUE THREAD

**Issue by serialpotato** [rclone/rclone#5855]
Hi, A certain local provider has just launched a cloud service. How do I go find about its compatibility with rclone, or add it here? 

From the app logs, I see following details:

```
Performing login to server https://cloud.jazzdrive.com.pk
Resetting validation key
Setting OAuth2 tokens [abcdefgh-vuwxyz-1234-56789-12213232] XaXbXcXd,XeXfXgXh,XiXeXjXk
Performing oauth2 login with url: https://cloud.jazzdrive.com.pk/sapi/login?action=login&responsetime=true
Performing request [abcdefgh-vuwxyz-1234-56789-12213232] https://cloud.jazzdrive.com.pk/sapi/login?action=login&responsetime=true
Refreshed OAuth2 tokens [abcdefgh-vuwxyz-1234-56789-12213232] XaXbXcXdXeXfXgXhXiXe
Updating session ID and validation key
Setting OAuth2 tokens [lmnopq-rstuvwz-112233-445577] XuXvXwXyXz, XwXtXnXo, XqXrXsXt
Performing request [lmnopq-rstuvwz-112233-445577] https://cloud.jazzdrive.com.pk/sapi/profile?action=get
Login ended with result 0
Performing upgrade actions to handle root folder selection
Unable to setup root folder path during upgrade
Performing post login actions...
Setting OAuth2 tokens [abcdefgh-vuwxyz-1234-56789-12213232] XaXbXcXdXeXfXgXhXiXe
Performing request [abcdefgh-vuwxyz-1234-56789-12213232] https://cloud.jazzdrive.com.pk/sapi/subscription/plan?action=get
Setting OAuth2 tokens [abcdefgh-vuwxyz-1234-56789-12213232] XaXbXcXdXeXfXgXhXiXe
Performing request [abcdefgh-vuwxyz-1234-56789-12213232] https://cloud.jazzdrive.com.pk/sapi/subscription?action=get
Performing magic and mobile and web uploads folder creation
Setting OAuth2 tokens [abcdefgh-vuwxyz-1234-56789-12213232] XaXbXcXdXeXfXgXhXiXe
Performing request [abcdefgh-vuwxyz-1234-56789-12213232] https://cloud.jazzdrive.com.pk/sapi/media/folder/root?action=get
Magic folder setup success
Scanning remote folders
Getting list of remote folders
Setting OAuth2 tokens [abcdefgh-vuwxyz-1234-56789-12213232] XaXbXcXdXe, XaXbXcXdXe, XaXbXcXdXe
Performing request [abcdefgh-vuwxyz-1234-56789-12213232] https://cloud.jazzdrive.com.pk/sapi/profile/changes?action=get&from=12345678U000000Y&type=folder&responsetime=true&sortby=creationdate&sortorder=descending
Changed items: {"requesttime":163817150232,"responsetime":1638172323,"data":{"folder":{"N":[844486,844377]}}}
There are NEW remote items for source folder
Getting folder items by ids...
Getting 2 items from Server (paging offset=0, limit=300)
Setting OAuth2 tokens [abcdefgh-vuwxyz-1234-56789-12213232] XaXbXcXdXe, XaXbXcXdXe, XaXbXcXdXe
Performing request [abcdefgh-vuwxyz-1234-56789-12213232] https://cloud.jazzdrive.com.pk/sapi/media/folder?action=get
Post login actions completed with success
Show wizard for first run
Scanning subfolders of: a1b2c3d4e5f6/ a1b2c3d4e5f6/ a1b2c3d4e5f6/ a1b2c3d4e5f6 ...
Root folder: a1b2c3d4e5f6
Local folders: [a1b2c3d4e5f6/a1b2c3d4e5f6/a1b2c3d4e5f6/a1b2c3d4e5f6]
Setting selected folders: [C:/Users/Admin/Jazz Drive, C:/Users/Admin/Jazz Drive/abc]
Contacts sync disabled
Entering Offline state from Logout
Setting up user depended services
Initializing store folder_items
Initializing store media_items
Initializing store incoming_folder_items
Initializing store incoming_media_items
Loading history store with root folder a1b2c3d4e5f6/a1b2c3d4e5f6/a1b2c3d4e5f6/a1b2c3d4e5f6
Reading 0 entries from history store
Setting sync tag service message handler
Start listening toJazzDrive.SyncTag
Tools action listener init
Start network status checker
Notify offline state
Performing request [abcdefgh-vuwxyz-1234-56789-12213232] https://cloud.jazzdrive.com.pk/sapi/system/information?action=get
Tray icon state changed to offline
Entering Online state from Offline
Start local changes check
Unblock local updates detection
Notify controller set to online state
Tray icon state changed to online
Setting OAuth2 tokens [abcdefgh-vuwxyz-1234-56789-12213232] XaXbXcXdXe, XaXbXcXdXe, XaXbXcXdXe
Performing request [abcdefgh-vuwxyz-1234-56789-12213232] https://cloud.jazzdrive.com.pk/sapi/media?action=get-storage-space&softdeleted=true
Updating quota: 0; user used: 4613456789; group used: 4613456789 unlimited: 1
Start remote changes detection
Starting timer for remote changes polling with interval: 300 secs
[WARNING] QObject::startTimer: Timers cannot be started from another thread
[INFO] Performing profile check...
[INFO] Performing feature check...
[INFO] Starting upload status validation...
[INFO] Setting OAuth2 tokens [abcdefgh-vuwxyz-1234-56789-12213232] XaXbXcXdXe, XaXbXcXdXe, XaXbXcXdXe
[INFO] Performing request [abcdefgh-vuwxyz-1234-56789-12213232] https://cloud.jazzdrive.com.pk/sapi/profile?action=get
[INFO] Setting OAuth2 tokens [abcdefgh-vuwxyz-1234-56789-12213232] XaXbXcXdXe, XaXbXcXdXe, XaXbXcXdXe
[INFO] Performing request [abcdefgh-vuwxyz-1234-56789-12213232] https://cloud.jazzdrive.com.pk/sapi/features?validationkey=3684853bc932329c78343434
No items for upload status validation. Service not started
Start sync notification listener
Stopping network status checker...
Block local updates detection
Block local contacts updates detection
Stopping remote changes checks...
Stopping upload status validation...
Refresh started
Notifying email validation status: valid
[INFO] Feature check completed : {"data":{"features":[{"name":"securelinks","status":"notactivatable","deactivatablebyuser":false,"packages":["pro"]},{"name":"pdfviewer","status":"active","deactivatablebyuser":false,"packages":["pro"]},{"name":"desktop","status":"active","deactivatablebyuser":false,"packages":[]}]},"responsetime":1638171543148}
[INFO] Notify sync status: First refresh start
[INFO] Tray icon state changed to syncing
[INFO] Notify filesystem refresh start
[INFO] Notify getting updates: 0/0
[INFO] Notify getting updates: 0/0
[INFO] Notify getting updates: 0/0
[INFO] Notify reconcile: 0/0
[INFO] Adding root to sync tag cache
[INFO] Notify sync status: Propagate change start
[INFO] Notify downloading item: 43ew434sd9394.jpg
[INFO] Notify downloading item: 43ew434sd9394.jpg
[INFO] Notify downloaded item: 43ew434sd9394.jpg
[INFO] Notify history event: operation: LocalAdd, path: a1b2c3d4e5f6/a1b2c3d4e5f6/a1b2c3d4e5f6/a1b2c3d4e5f6.jpg
[INFO] Adding history event for item with guid 176908647
[INFO] Setting OAuth2 tokens [abcdefgh-vuwxyz-1234-56789-12213232] XaXbXcXdXe, XaXbXcXdXe, XaXbXcXdXe
[INFO] Performing request [abcdefgh-vuwxyz-1234-56789-12213232] https://cloud.jazzdrive.com.pk/sapi/download/thumbnail?action=get&fdoid=18483493&type=s
[INFO] Notify downloading item: fdf9sas02ewe.jpg
[INFO] Notify downloaded item: fdf9sas02ewe.jpg
```

P.S- All the names/serials/hashes have been changed.

I login using OTP. There is no password option for the app. It only has phone number, which upon providing requires OTP.

**Comment by animosity22** [comment]
They have to have an API to be requested and you can request a few backend and if enough people vote for it or you want to implement it, you can do that as well as submit a PR.

https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#writing-a-new-backend

If you find those out or decide, submit a new issue with the proper details.

Thanks!
